code/modules/hydroponics/seed_mobs.dm

```diff
@@ -57,6 +57,9 @@
 		O = pick(currently_querying)
 		while(currently_querying.len && !check_observer(O)) //While we the list has something and
 			currently_querying -= O				//Remove them from the list if they don't get checked properly
+			if(currently_querying.len) // Are we out of people?
```

Comment:
I think this is meant to be a repeating loop for querying users to become a diona, this break if it has no length will only make it loop once effectively not making it loop at all.